### PR TITLE
Prometheus: Query builder UX tweaks and feedback link

### PR DESCRIPTION
**Pull request by torkelo**
* Move the query patterns & Preview toggle to the left  (To align with Big query data source , and cloudwatch data source) 
* Rename Preview to `Raw query`  (I will rename model property also after ok) 
* Introduce a Give feedback link 

Screenshot from 2022-04-12 19-25-47

When you hover over the link it turns link blue (and has tooltip that says "This query builder is new, please let us know how we can improve it") 

* [x] Update editorPreview model property to rawQuery
* [x] Create GitHub discussion issue to link to for feedback

### CODE REVIEW

**Comment by torkelo** [comment]
Alt design, where we move the beta badge from the editor mode radio toggle and but it in next to the feedback link. 
Screenshot from 2022-04-12 19-14-22

File: public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.tsx [hunk 1]
```diff
@@ -95,14 +86,22 @@ export const PromQueryEditorSelector = React.memo<PromQueryEditorProps>((props)
...
           </>
         )}
-        <QueryHeaderSwitch
-          label="Preview"
-          value={query.editorPreview}
-          onChange={onQueryPreviewChange}
-          disabled={editorMode !== QueryEditorMode.Builder}
-        />
+        {editorMode === QueryEditorMode.Builder && (
+          <BetaBadge feedbackUrl="https://github.com/grafana/grafana/pull/47653" />
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.tsx:93]: The url should be probably to new bug report, right? 
```suggestion
          <BetaBadge feedbackUrl="https://github.com/grafana/grafana/issues/new?assignees=&labels=type%3A+bug&template=1-bug_report.md" />
```

**Comment by diegoadams** [comment]
My only concern is the placement of the Beta label & Give Feedback action. It feels like it conflicts with the query-specific actions (patterns/raw query).  

As the data source determines these new features, what about having the beta / give feedback action at the top next to the data source. This way, we don't have to repeat the Beta label on every query editor.

The design is not to scale, but it could look like this. 
Screen Shot 2022-04-13 at 10 39 12 AM



Screen Shot 2022-04-13 at 10 39 20 AM

**Comment by diegoadams** [comment]
I like the beta badge being present on the page. As a user, I want to know when an app or feature is in beta so that I can properly set my expectations on its current state and future development prospects.

**Comment by torkelo** [comment]
@diegoadams 

> My only concern is the placement of the Beta label & Give Feedback action. It feels like it conflicts with the query-specific actions (patterns/raw query). 

Not sure it conflicts with those actions? In what way? They all relate to the query builder. The feedback option is only shown when you have the builder mode enabled. 

We cannot show it where you suggest as it would make users think the whole data source or Explore page is a beta feature and that is the thing you give feedback to. The reason it's inside the query editor is that it's only for the new query builder.

**Comment by diegoadams** [comment]
@torkelo , would the feedback link open a modal where the user can submit their feedback?

File: public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.tsx:93]: @ivanahuckova 
no it should go to a github dissusion issue on the feature, that describe the feature, current limitations, our planned/hoped improvements. And asks for feedback. 

Example for what Github links to for the PR Review files pane link:

Screenshot from 2022-04-12 15-24-30

 https://github.com/github/feedback/discussions/12341

**Comment by torkelo** [comment]
@diegoadams no, it would link to a GitHub discussion issue that describe the Prometheus query builder, what problem it tries to solve, it's current limitations, and currently planned improvements, special areas of feedback (if any). 

Screenshot from 2022-04-12 15-24-30

Example issue: 
https://github.com/github/feedback/discussions/12341

**Comment by torkelo** [comment]
The link would open in a new tab

**Comment by torkelo** [comment]
@tumaranja thanks, yea excited to try this. Hope it works out, either way we learn something :)

**Comment by aocenas** [comment]
@diegoadams for the link position, there is also no such place for dashboards. Also in dashboards, you can have a mix of different data sources in one panel (and will be possible also for explore at some point) so we have to keep it in the query editor itself.

File: conf/defaults.ini [hunk 2]
```diff
@@ -230,6 +230,9 @@ application_insights_connection_string =
 # Optional. Specifies an Application Insights endpoint URL where the endpoint string is wrapped in backticks ``.
 application_insights_endpoint_url =
 
+# Controls if the UI contains any links to user feedback forms
+feedback_links_enabled = true
```
Comment (aocenas) [inline conf/defaults.ini:234]: Does it make sense to make this global? What is the usecase to disable it?

**Review by ivanahuckova — COMMENTED** [review]
Looks good! 

1 question, would it make sense to add `config.feedbackLinksEnabled` handling directly into `<FeedbackLink />`? That way we would only show  FeedbackLinks when they are enabled and we wouldn't have to think about checking it at every place where we use it.

**Review by aocenas — APPROVED** [review]

File: conf/defaults.ini (hunk 2 above)
Comment (torkelo) [reply conf/defaults.ini:234]: Mostly a backup in case CX get's complaints from customers who do not want this link

**Pull request merged**